paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentConfirmationMediator.kt

```diff
@@ -0,0 +1,159 @@
+                    "Parameter type of '${option::class.simpleName}' cannot be used with " +
+                        "${this::class.simpleName} to read a result"
+                ),
+                message = R.string.stripe_something_went_wrong.resolvableString,
+                errorType = PaymentConfirmationErrorType.Internal,
+            )
+
+        return when (val action = definition.action(confirmationOption, intent)) {
+            is PaymentConfirmationDefinition.ConfirmationAction.Launch -> {
+                launcher?.let {
+                    Action.Launch(
+                        launch = {
+                            persistedParameters = Parameters(
+                                confirmationOption = confirmationOption,
+                                intent = intent,
+                                deferredIntentConfirmationType = action.deferredIntentConfirmationType,
+                            )
+
+                            definition.launch(
+                                launcher = it,
+                                arguments = action.launcherArguments,
+                                confirmationOption = confirmationOption,
+                                intent = intent,
+                            )
+                        },
+                    )
+                } ?: run {
+                    val exception = IllegalStateException(
+                        "No launcher for ${definition::class.simpleName} was found, did you call register?"
+                    )
+
+                    Action.Fail(
+                        cause = exception,
+                        message = exception.stripeErrorMessage(),
+                        errorType = PaymentConfirmationErrorType.Fatal,
+                    )
+                }
+            }
+            is PaymentConfirmationDefinition.ConfirmationAction.Complete -> {
+                Action.Complete(
+                    intent = action.intent,
+                    confirmationOption = action.confirmationOption,
+                    deferredIntentConfirmationType = action.deferredIntentConfirmationType,
+                )
+            }
+            is PaymentConfirmationDefinition.ConfirmationAction.Fail -> {
+                Action.Fail(
+                    cause = action.cause,
+                    message = action.message,
+                    errorType = action.errorType,
+                )
+            }
+        }
+    }
+
+    sealed interface Action {
+        class Launch(
+            val launch: () -> Unit,
+        ) : Action
+
+        data class Fail(
+            val cause: Throwable,
+            val message: ResolvableString,
+            val errorType: PaymentConfirmationErrorType,
+        ) : Action
+
+        data class Complete(
+            val intent: StripeIntent,
+            val confirmationOption: PaymentConfirmationOption,
+            val deferredIntentConfirmationType: DeferredIntentConfirmationType? = null,
```

Comment:
@jaynewstrom-stripe I think you mentioned potentially injecting this attribute. I could have another mediator object that is shared between `PaymentSheet` and `IntentConfirmationDefinition` that is able to push this value into the object for `PaymentSheet` to grab afterwards.

Other option is to have `PaymentConfirmationMetadata where we can store attributes like this and use.